pkg/metrics/collector.go

```diff
@@ -0,0 +1,176 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	cv "github.com/openshift/managed-upgrade-operator/pkg/clusterversion"
+	ucm "github.com/openshift/managed-upgrade-operator/pkg/upgradeconfigmanager"
+)
+
+var (
```

Comment:
Please add metrics tag to each metric. 
tag ref: https://github.com/openshift/managed-upgrade-operator/blob/361eebdd01f2bf57099e6e776242d04a87277ea0/pkg/metrics/metrics.go#L21
implemenation ref: https://github.com/openshift/managed-upgrade-operator/blob/361eebdd01f2bf57099e6e776242d04a87277ea0/pkg/metrics/metrics.go#L105-L109